### PR TITLE
Aave lending

### PR DESCRIPTION
**Pull request by ylv-io**
# Description

* Adds a LendAndBridgeWorkflow contract for lending and bridging assets.
* Adds external Aave Pool interface IAavePool for integration.
* Modifies Viewer.sol to include Aave data retrieval functionality.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Deployment
- [ ] Forge Script
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: src/viewers/Viewer.sol [hunk 1]
```diff
@@ -51,4 +70,69 @@ contract Viewer is Initializable, UUPSUpgradeable, OwnableUpgradeable {
...
+        returns (AaveUserData[] memory metrics)
+    {
+        IAavePool pool = IAavePool(poolAddressProvider.getLendingPool());
+        metrics = new AaveUserData[](assets.length);
+
+        for (uint256 i = 0; i < assets.length; i++) {
+            IAavePool.ReserveDataLegacy memory reserveData = pool.getReserveData(assets[i]);
+            IAavePool.ReserveConfigurationMap memory config = pool.getConfiguration(assets[i]);
+
+            uint256 supplyBalance = IERC20(reserveData.aTokenAddress).balanceOf(kintoWallet);
```
[the remark this answers is not in the document]
Comment (ylv-io) [reply src/viewers/Viewer.sol:93]: Atoken are 1 to 1 to base tokens, so this should be the same value, but I'll make sure to check in the tests.

File: src/access/workflows/LendAndBridgeWorkflow.sol [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import {SafeERC20} from "@openzeppelin-5.0.1/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin-5.0.1/contracts/utils/Address.sol";
+import {IERC4626} from "@openzeppelin-5.0.1/contracts/interfaces/IERC4626.sol";
+
+import {IAavePool, IPoolAddressesProvider, IStaticATokenFactory} from "@kinto-core/interfaces/external/IAavePool.sol";
+import {IBridge} from "@kinto-core/interfaces/bridger/IBridge.sol";
+import {IAccessPoint} from "@kinto-core/interfaces/IAccessPoint.sol";
+import {IBridger} from "@kinto-core/interfaces/bridger/IBridger.sol";
+
+contract LendAndBridgeWorkflow {
```
Comment (rrecuero) [inline src/access/workflows/LendAndBridgeWorkflow.sol:14]: This is not needed anymore right? We can remove this workflow

File: src/access/workflows/AaveBorrowWorkflow.sol [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @notice Borrows an asset from Aave
+     * @param asset The address of the asset to borrow
+     * @param amount The amount to borrow
+     */
+    function borrow(address asset, uint256 amount) external {
```
Comment (rrecuero) [inline src/access/workflows/AaveBorrowWorkflow.sol:41]: This should be borrow and bridge

File: src/access/workflows/LendAndBridgeWorkflow.sol (hunk 2 above)
Comment (ylv-io) [reply src/access/workflows/LendAndBridgeWorkflow.sol:14]: Yes. This is not needed.

File: src/access/workflows/AaveRepayWorkflow.sol [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     * @param asset The address of the borrowed asset to repay
+     * @param amount The amount to repay (use type(uint256).max for max debt)
+     * @param onBehalfOf The address of the user who will get their debt reduced
+     * @return amountRepaid The actual amount repaid
+     */
+    function repay(address asset, uint256 amount, address onBehalfOf) external returns (uint256) {
+        address pool = poolAddressProvider.getPool();
+
+        // If amount is max uint256, get the debt for this specific asset
+        if (amount == type(uint256).max) {
```
Comment (rrecuero) [inline src/access/workflows/AaveRepayWorkflow.sol:47]: should we use 0 instead of max?

**Review by rrecuero — APPROVED** [review]

**Pull request merged**